### PR TITLE
Sd hk firebase

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firebase-hooks`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

Notes:

* Don't do too much here. In past, pairs have built the whole add item form in the first week. The best approach is just to make a throw away view with a `<button/>` and a `<div/>`. Click the `<button/>` to send some data to the database. Have the data show up in the `<div/>`. Simple!
* Also, if you need more direction on how to use `react-firebase-hooks`, you can reference this commit from Alejandro: https://github.com/alejandronanez/smart-shopping-list-demo/commit/d590adc3224872e71d000c41977a1b7a84542c31 and/or watch this Loom video where he explains how it works: https://www.loom.com/share/0dffa1a63bf24b449d6565c76606bbd9

### PR DESCRIPTION
**Pull request by sandaiiyahh**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- We installed `firebase` & `react-firebase-hooks`
- We used firestore to add an item in the database. We created our first item, "egg"
- We created 2 components (`ItemsList` and `AddItem`) and imported them to our App
- We created a template for our items. For now, the template is just "name" and "date" but we plan to also include "id" and "frequency", for how often we purchase the item
- We used `useCollection` from `react-firebase-hooks` to display the data onto the UI
- Clicking the "Add an item" button will create a new item in the database

Documentation on `react-firebase-hooks` can be found [here](https://github.com/CSFrequency/react-firebase-hooks/blob/master/firestore/README.md#usecollection)

## Related Issue

#1 

## Acceptance Criteria

- [ ] The following have been added as project dependencies: firebase & react-firestore
- [ ] Make a change in the Firestore database and it shows up in the app
- [ ] Make a change in the app and it shows up in the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

### After

Screen Shot 2021-07-11 at 4 29 21 PM

## Testing Steps / QA Criteria

- From your terminal, pull down this branch
- `npm install`
- `npm start`
- See if data from our `items` collection appears on the app
- Click on the "Add an item" button and a new item should appear
- Check to see if the new item also appears in the `items` collection of our database [here](https://console.firebase.google.com/u/0/project/tcl-28-shopping-list/firestore/data/~2Fitems~2FOoB4YGCWIAZU3qHoXJwz)

### CODE REVIEW

**Review by isaabutaa — COMMENTED** [review]
This looks great! I'm not too familiar with firebase, but I can see that in the AddItem component you're adding new items to the collection with the button click, and in the ItemsList component, you're mapping through the collection and rendering the items. When I view the app locally, the data persists.

I can see there being a potential merge conflict with the ks-ia-router branch, but not because of anything anyone did wrong. 😊 Really nicely done! 👏

**Review by ksiman14 — APPROVED** [review]
All looks good. I am also not familiar with Firebase and am excited to learn about what you all did when we get together this weekend! As Isa noted, this may create conflicts with some of our changes, but not because anything is incorrect.

File: src/AddItem.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const AddItem = () => {
+  // get a reference to the Firestore document
+  const handleClick = async () => {
+    const itemTemplate = {
+      name: 'New item',
+      date: Date.now(),
+    };
+    await firestore.collection('items').add(itemTemplate);
+  };
```
Comment (luisaugusto) [inline src/AddItem.js:11]: I like the use of `async/await` here 😎

File: src/ItemsList.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { firestore } from './lib/firebase';
+import { useCollection } from 'react-firebase-hooks/firestore';
+
+const ItemsList = () => {
+  const [snapshot, loading, error] = useCollection(
+    firestore.collection('items'),
+  );
```
Comment (luisaugusto) [inline src/ItemsList.js:7]: Nice job getting the Firebase hooks setup! 🔥

**Review by luisaugusto — APPROVED** [review]
Great job, I know this first issue can be a little challenging to get the intially firebase configurations, but looks like you got a handle on it! Feel free to merge this in when you are ready 🥳

**Pull request merged**